append.c

```diff
@@ -554,3 +557,28 @@ void	loc_vfprintf(FILE *file, const char *format, va_list args)
   buf_p = append_vformat(buf, limit, format, args);
   fwrite(buf, 1, (buf_p - buf), file);
 }
+
+/*
+ * Local implementation of dprintf so we can use %p and other non-standard formats.
+ */
+void	loc_dprintf(int fd, const char *format, ...)
+{
+  va_list args;
+  va_start(args, format);
+  loc_vdprintf(fd, format, args);
+  va_end(args);
+}
+
+/*
+ * Local implementation of vdprintf so we can use %p and other non-standard formats.
+ */
+void	loc_vdprintf(int fd, const char *format, va_list args)
+{
+  // these are simple messages so this limit is ok
+  char buf[256];
+  char *buf_p, *limit;
+
+  limit = buf + sizeof(buf);
+  buf_p = append_vformat(buf, limit, format, args);
+  (void)!write(fd, buf, buf_p - buf);
```

Comment:
Please remove the ! before the write.

Reply:
! is required to avoid the warning with the new gcc version:
```
append.c:583:11: warning: ignoring return value of ‘write’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  583 |     (void)write(fd, buf, buf_p - buf);
```
Here has more discussion:
https://stackoverflow.com/questions/7271939/warning-ignoring-return-value-of-scanf-declared-with-attribute-warn-unused-r